pkg/controller/assignmetadata/assignmetadata_controller.go

```diff
@@ -0,0 +1,146 @@
+	"github.com/open-policy-agent/gatekeeper/pkg/logging"
+	"github.com/open-policy-agent/gatekeeper/pkg/readiness"
+	"github.com/open-policy-agent/gatekeeper/pkg/util"
+	"github.com/open-policy-agent/gatekeeper/pkg/watch"
+	//"github.com/open-policy-agent/gatekeeper/pkg/mutation"
+
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var (
+	log = logf.Log.WithName("controller").WithValues(logging.Process, "assignmentmetadata_controller")
+)
+
+type Adder struct {
+	Opa              *opa.Client
+	WatchManager     *watch.Manager
+	ControllerSwitch *watch.ControllerSwitch
+	Tracker          *readiness.Tracker
+}
+
+// Add creates a new AssignMetadata Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func (a *Adder) Add(mgr manager.Manager) error {
+
+	r := newReconciler(mgr)
+	events := make(chan event.GenericEvent, 1024)
+	return add(mgr, r, events)
+}
+
+func (a *Adder) InjectOpa(o *opa.Client) {
+	a.Opa = o
+}
+
+func (a *Adder) InjectWatchManager(w *watch.Manager) {
+	a.WatchManager = w
+}
+
+func (a *Adder) InjectControllerSwitch(cs *watch.ControllerSwitch) {
+	a.ControllerSwitch = cs
+}
+
+func (a *Adder) InjectTracker(t *readiness.Tracker) {
+	a.Tracker = t
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) *AssignMetadataReconciler {
+	r := &AssignMetadataReconciler{}
+	return r
+}
+
+type mapper struct {
+	packer util.EventPacker
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler, events <-chan event.GenericEvent) error {
+	// Create a new controller
+	c, err := controller.New("assignmetadata-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to the provided constraint
+	err = c.Watch(
+		&source.Channel{
+			Source:         events,
+			DestBufferSize: 1024,
+		},
+		&handler.EnqueueRequestsFromMapFunc{ToRequests: util.EventPacker{}},
+	)
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to ConstraintStatus
+	err = c.Watch(
+		&source.Kind{Type: &mutationsv1alpha1.AssignMetadata{}},
+		&handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// AssignMetadataReconciler reconciles a AssignMetadata object
+type AssignMetadataReconciler struct {
+	//system *mutation.System // A Mutator cache, introduced in PR: https://github.com/open-policy-agent/gatekeeper/pull/928/
+}
+
+// +kubebuilder:rbac:groups=assignmetadata.gatekeeper.sh,resources=*,verbs=get;list;watch;create;update;patch;delete
```

Comment:
The groups should be `mutation.gatekeeper.sh`

Reply:
also, after running make manifests we should push the updated role.yaml as well for consistency.

Reply:
Done